### ISSUE TITLE
race condition fix

### ISSUE THREAD

**Issue by remdragon** [niccokunzmann/hanging_threads#89]
get_current_frames() can crash because the list of active threads can change between calling threading.enumerate() and later sys._current_frames()

Following is one way to refactor to avoid that crash. Basically, the thread has to exist in both sets of data to qualify to be returned from the function

```
def get_current_frames() -> Dict[int,Dict[str,Any]]:
    """Return current threads prepared for 
    further processing.
    """
    threads = {thread.ident: thread for thread in threading.enumerate()}
    frames = {}
    for thread_id, frame in sys._current_frames().items():
        thread = threads.get ( thread_id, None )
        if thread is not None:
            frames[thread_id] = {
                'frame': thread2list(frame),
                'time': None,
                'id': thread_id,
                'name': thread.name,
                'object': thread
            }
    return frames
```

**Comment by niccokunzmann** [comment]
This is fixed in #91.

**Comment by niccokunzmann** [comment]
Thanks, this should be uploaded as v2.0.6 https://pypi.org/project/hanging-threads/
